WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/UiModelMapper.kt

```diff
@@ -76,6 +78,48 @@ class UiModelMapper
         return mapStatsWithOverview(TimeStatsType.OVERVIEW, useCaseModels, showError)
     }
 
+    @Suppress("CyclomaticComplexMethod")
+    fun mapSubscribers(useCaseModels: List<UseCaseModel>, showError: (Int) -> Unit): UiModel {
+        val allFailing = useCaseModels.isNotEmpty() && useCaseModels.fold(true) { acc, useCaseModel ->
+            acc && useCaseModel.state == ERROR
+        }
+        val allFailingWithoutData = useCaseModels.fold(true) { acc, useCaseModel ->
+            acc && useCaseModel.state == ERROR && useCaseModel.data == null
+        }
```

Comment:
This is being used in both `mapInsights()` and in `mapSubscribers()` for example, and could probably be brought out into their own short functions. What do you think?

Reply:
Done: fd93f164acf52e6f3770e15eb9ac14b76e62a7db